### PR TITLE
fix(condo): DOMA-5218 registerResident mutation updated

### PR DESCRIPTION
**Pull request by AleX83Xpert**
Additionally, search existing resident by address key.

### CODE REVIEW

File: apps/condo/domains/resident/schema/RegisterResidentService.js [hunk 1]
```diff
@@ -35,8 +41,20 @@ const RegisterResidentService = new GQLCustomSchema('RegisterResidentService', {
                     unitType,
                     user: { connect: { id: context.authedItem.id } },
                 }
+
+                // TODO(AleX83Xpert): Think about how to prevent copying this condition :-/
+                const client = conf.NODE_ENV === 'test' || get(conf, 'ADDRESS_SERVICE_CLIENT_MODE') === 'fake'
```
Comment (sitozzz) [inline apps/condo/domains/resident/schema/RegisterResidentService.js:46]: What if client can't be found in conf? Should this case been handled with an special error?

Comment (AleX83Xpert) [reply apps/condo/domains/resident/schema/RegisterResidentService.js:46]: Resident and Property models can't work without address service and without config field. If there is no config for address service, property and resident creation will fail.

**Review by sitozzz — APPROVED** [review]

File: apps/condo/domains/resident/schema/RegisterResidentService.js (hunk 1 above)
Comment (dkoviazin) [inline apps/condo/domains/resident/schema/RegisterResidentService.js:46]: You can move this logic to createAddressServiceClientInstance function with all checks for NODE_ENV and ADDRESS_SERVICE_CLIENT_MODE
Reasons:
1. RegisterResidentService will have less code and checks (for address service environment)
2. Now you  need to copy-paste this logic to every place where you is using addressService

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/resident/schema/RegisterResidentService.js (hunk 1 above)
Comment (AleX83Xpert) [reply apps/condo/domains/resident/schema/RegisterResidentService.js:46]: Unfortunately, functions have different signatures :(
The real client needs only the address service URL. The test client needs an object.

**Review by Alllex202 — APPROVED** [review]

File: apps/condo/domains/resident/schema/RegisterResidentService.js (hunk 1 above)
Comment (pahaz) [reply apps/condo/domains/resident/schema/RegisterResidentService.js:46]: But you can change it

**Review by pahaz — APPROVED** [review]

**Pull request merged**